wg-naming/README.md

```diff
@@ -0,0 +1,70 @@
+<!---
+This is an autogenerated file!
+
+Please do not edit this file directly, but instead make changes to the
+sigs.yaml file in the project root.
+
+To understand how this file is generated, see https://git.k8s.io/community/generator/README.md
+--->
+# Naming Working Group
+
+
+## Stakeholder SIGs
+* SIG Architecture
+* SIG Contributor Experience
+* SIG Docs
+
+
+
+## Organizers
+
+* Celeste Horgan (**[@celestehorgan](https://github.com/celestehorgan)**), CNCF
+* Stephen Augustus (**[@justaugustus](https://github.com/justaugustus)**), VMware
+* Zach Corleissen (**[@zacharysarah](https://github.com/zacharysarah)**), Linux Foundation
+
+## Contact
+- Slack: [#wg-naming](https://kubernetes.slack.com/messages/wg-naming)
+- [Mailing list](https://groups.google.com/forum/#!forum/kubernetes-wg-naming)
+- [Open Community Issues/PRs](https://github.com/kubernetes/community/labels/wg%2Fnaming)
+<!-- BEGIN CUSTOM CONTENT -->
+
+**The following section will be reworked and formalized as a charter once the
+Working Group has been approved by the Steering Committee.**
+
+### Goals
+
+- Evaluate language and naming choices within the Kubernetes organization, with
+  a specific initial focus of:
+  - Removing language which cannot be fully separated from racist, sexist or
+    homophobic connotations in cultural contexts outside of computer
+    programming
+  - Improving clarity of codebases and documentation by replacing idioms,
+    metaphors and slang specific to the English language
+- Create a list of terms to avoid and propose alternatives
+- Define a process by which any member of the Kubernetes organization can
+  propose a language recommendation, have them evaluated based on a rubric, and
+  see them implemented across all codebases.
+  - Provide an easily findable location for language recommendations and
+    follow-up issues, similar to an architectural decision record.
+- Work with stakeholder SIGs to implement the changes recommended. We
+  anticipate the following:
+  - Provide stakeholder SIGs with guidance on naming, language conventions, and
+    processes
+  - Collaborate with SIG Architecture and other stakeholders on an
+    implementation timeline and strategies for dealing with follow-up issues
+    from renaming, like deprecations
+  - Collaborate with SIG Docs and SIG Contributor Experience on documenting
+    language recommendations and processes
+  - Work with the Code of Conduct Committee to add code architecture to the COC
+
+### Dissolution Criteria
+
+When this WG dissolves, the Kubernetes community should have:
+
+- A process in place to evaluate language changes on an ongoing basis
+- A list of terms to avoid in codebases across the organization
+- A timeline on which to replace component names in the kubernetes/kubernetes
+  codebase
+- A revised Code of Conduct including the list of terms to avoid
```

Comment:
Our code of conduct is tied to the Contributor Covenant, so my feeling is we don't add this explicitly to the CoC itself, and instead offer guidance in the areas of documentation where it's appropriate. If someone continues using terminology we've eliminated from the project, we should provide guidance and evaluate on a case-by-case basis. It took a non-trivial amount of time to change terminology in the past, so we should expect a transition time.

Reply:
As @jdumars mentions, we currently source text from the CNCF CoC policy (which is based on Contributor Covenant). If we find ways we can strengthen or clarify it, it's not impossible to do -- even advocating for those changes CNCF wide if need be. That said, I wouldn't say this is an explicit must as far as exit criteria for the WG, to modify it. It would be a larger undertaking. So perhaps strike this as a formal goal, and revisit with the CoCC as needed?

Reply:
@jdumars I think not embedding it in the COC is reasonable, however I want to emphasize that we _don't_ want to provide just provide guidance – we want to provide _binding_ guidance.

Initiatives like this one often become wasted time because it's easy to produce guidance but hard to follow-through. The initial idea behind enshrining these requirements in the COC was to make them, well _required_, rather than suggested.

I'm happy to leave the format/final home fuzzy, but I would emphasize that the WG's recommendations are expected to be followed through on and adhered to in future.

Maybe..

```suggestion

```

Strike this line, and edit [this one?](https://github.com/kubernetes/community/pull/4884/files#diff-bbd3d7bbc2884b2628bf43deea5e4391R65)

Reply:
Given that the CoC already covers inappropriate or harassing language in code and documentation, the proposed result here is for this WG to _extend_ or _specify_ the list of terms or conditions when the CoC Committee would be called to act. In other words, this WG proposal, if it accomplishes its goal (and, personally, I think this is a great goal) would add additional work for the CoC Committee (at least under the current charter).

With that in mind, I would like to propose that the CoC Committee be considered a stakeholder SIG and one of our five members be added to the initial WG-leads list for WG-naming. I think that this WG and the CoCC work hand-in-hand to craft this language and then to roll out or advise on the implementation path.

Reply:
I am volunteering to represent the committee for this work.